File: insert_csrf.py
import os
import re

views_dir = os.path.join(os.path.dirname(__file__), 'app', 'views')
count = 0

# Regex to match form tags with method="POST" (case-insensitive)
pattern = re.compile(r'(<form\b[^>]*method=[\'"]?POST[\'"]?[^>]*>)', re.IGNORECASE)

replacement = r'\1\n    <input type="hidden" name="_csrf_token" value="<?= Session::generateCsrfToken() ?>">'

for root, dirs, files in os.walk(views_dir):
    for file in files:
        if file.endswith('.php'):
            filepath = os.path.join(root, file)
            with open(filepath, 'r', encoding='utf-8') as f:
                content = f.read()
            
            # Check if token is already present to avoid duplication
            if '_csrf_token' not in content:
                new_content = pattern.sub(replacement, content)
                if new_content != content:
                    with open(filepath, 'w', encoding='utf-8') as f:
                        f.write(new_content)
                    print(f"Updated: {filepath}")
                    count += 1

print(f"Total files updated: {count}")

File: insert_csrf.php
<?php
$dir = __DIR__ . '/app/views';

$iterator = new RecursiveIteratorIterator(new RecursiveDirectoryIterator($dir));
$count = 0;

foreach ($iterator as $file) {
    if ($file->isFile() && $file->getExtension() === 'php') {
        $content = file_get_contents($file->getPathname());
        
        // Match <form ... method="POST" ...> or method="post"
        $pattern = '/(<form\b[^>]*method=[\'"]POST[\'"][^>]*>)/i';
        $replacement = "$1\n    <input type=\"hidden\" name=\"_csrf_token\" value=\"<?= Session::generateCsrfToken() ?>\">";
        
        $newContent = preg_replace($pattern, $replacement, $content);
        
        // Also check if some forms use JS to submit, we might need to handle them, but looking at the grep, they mostly have method="POST"
        // Except editUserForm which has method="POST" id="editUserForm"
        // Also processForm in detail.php which has no method, it's submitted via JS. It might use GET or POST.
        
        if ($newContent !== $content) {
            file_put_contents($file->getPathname(), $newContent);
            echo "Updated: " . $file->getPathname() . "\n";
            $count++;
        }
    }
}

echo "Total files updated: $count\n";
